src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1857,7 +1859,7 @@ private bool IsTypeDeclarationStart()
             switch (this.CurrentToken.Kind)
             {
                 case SyntaxKind.ClassKeyword:
-                case SyntaxKind.DelegateKeyword:
+                case SyntaxKind.DelegateKeyword when PeekToken(1).Kind != SyntaxKind.AsteriskToken:
```

Comment:
did this just break the switch to be a linear search instead of normal integer-switch-dispatch?

Reply:
technically ever so slightly more work.  but i think it's much clearer to write: `when !IsFunctionPointerStart()`  up to you.

Reply:
No, adding `when` clauses doesn't slow things down.

---
In reply to: [348013181](https://github.com/dotnet/roslyn/pull/39879#discussion_r348013181) [](ancestors = 348013181)